File: ios/Parsers/CompanyParser.swift
import Foundation
import Crisp

@objc
public class CompanyParser: NSObject {

    @objc
    public static func fromDictionary(_ dict: NSDictionary) -> Company {
        let name = dict.getStringOrNil(forKey: "name")

        var url: URL? = nil
        if let urlString = dict.getStringOrNil(forKey: "url"), !urlString.isEmpty {
            url = URL(string: urlString)
        }

        let companyDescription = dict.getStringOrNil(forKey: "companyDescription")

        var employment: Employment? = nil
        if let employmentDict = dict.getDictionaryOrNil(forKey: "employment") {
            employment = EmploymentParser.fromDictionary(employmentDict)
        }

        var geolocation: Geolocation? = nil
        if let geolocationDict = dict.getDictionaryOrNil(forKey: "geolocation") {
            geolocation = GeolocationParser.fromDictionary(geolocationDict)
        }

        return Company(
            name: name,
            url: url,
            companyDescription: companyDescription,
            employment: employment,
            geolocation: geolocation
        )
    }
}

File: android/src/main/java/com/reactnativecrispchatsdk/parsers/CompanyParser.kt
package com.reactnativecrispchatsdk.parsers

import com.facebook.react.bridge.ReadableMap
import com.reactnativecrispchatsdk.getStringOrNull
import com.reactnativecrispchatsdk.getMapOrNull
import im.crisp.client.external.data.Company
import java.net.URL

/**
 * Parser for Company objects from JavaScript
 * Converts ReadableMap to Crisp SDK Company class
 *
 * TypeScript interface:
 * interface Company {
 *   name: string;
 *   url?: string;
 *   companyDescription?: string;
 *   employment?: Employment;
 *   geolocation?: Geolocation;
 * }
 */
object CompanyParser {
    fun fromReadableMap(map: ReadableMap): Company {
        val name = map.getString("name") ?: ""

        val url = map.getStringOrNull("url")?.takeIf { it.isNotEmpty() }?.let {
            try {
                URL(it)
            } catch (e: Exception) {
                null
            }
        }

        val companyDescription = map.getStringOrNull("companyDescription")

        val employment = map.getMapOrNull("employment")?.let {
            EmploymentParser.fromReadableMap(it)
        }

        val geolocation = map.getMapOrNull("geolocation")?.let {
            GeolocationParser.fromReadableMap(it)
        }

        return Company(name, url, companyDescription, employment, geolocation)
    }
}
